tests/system/python/packages/test_lab.py

```diff
@@ -0,0 +1,562 @@
+        else:
+            assert True
+
+    def test_event_toggled_sent_clear(self, foglamp_url, wait_time, retries):
+        print("Add sinusoid")
+        payload = {"name": "sin #1", "plugin": "sinusoid", "type": "south", "enabled": True}
+        post_url = "/foglamp/service"
+        resp = post_request(foglamp_url, post_url, payload)
+        assert resp["id"] != "", "ERROR! Failed to add sin #1"
+        assert resp["name"] != "", "ERROR! Failed to add sin #1"
+
+        print("Create event instance with threshold and asset; with notification trigger type toggled")
+        payload = {"name": "test #1", "description": "test notification instance", "rule": "Threshold", "channel": "asset",
+                "notification_type": "toggled", "enabled": True}
+        post_url = "/foglamp/notification"
+        post_request(foglamp_url, post_url, payload)
+        
+        get_url = "/foglamp/notification"
+        resp = get_request(foglamp_url, get_url)
+        for i in resp["notifications"] :
+            if "test" == i["name"]:
+                assert True
+
+        print("Set rule")
+        payload = {"asset": "sinusoid", "datapoint": "sinusoid", "trigger_value": "0.8"}
+        put_url = "/foglamp/category/ruletest%20%231"
+        put_request(foglamp_url, put_url, payload)
+        
+        get_url = "/foglamp/category/ruletest%20%231"
+        resp = get_request(foglamp_url, get_url)
+        assert len(resp) > 0
+
+        print("Set delivery")
+        payload = {"asset": "sin 0.8", "description": "asset notification", "enable": "true"}
+        put_url = "/foglamp/category/deliverytest%20%231"
+        put_request(foglamp_url, put_url, payload)
+
+        # TODO: FOGL-3115 verify asset tracker entry
+        s=wait_time*2
+        print("Verify sin 0.8 has been created")
+        while retries:
+            time.sleep(s)
+            get_url = "/foglamp/asset/sin%200.8"
+            resp = get_request(foglamp_url, get_url)
+            if len(resp) > 0:
+                assert True
+                break
+            retries -= 1
+
+        if retries == 0:
+            assert "TIMEOUT! sin 0.8 event not fired." + foglamp_url + "/foglamp/asset/sin%200.8"
+
+        print("When rule is triggred, There should be 2 entries in Logs->notifications NTFCL and NTFSN")
+        get_url = "/foglamp/audit?limit=1&source=NTFSN&severity=INFORMATION"
+        resp = get_request(foglamp_url, get_url)
+        assert len(resp['audit'])
+        for audit_detail in resp['audit']:
+            if "test #1" == audit_detail['details']['name']:
+                assert "INFORMATION" == audit_detail['severity']
+                assert "NTFSN" == audit_detail['source']
+        
+
+        get_url = "/foglamp/audit?limit=1&source=NTFCL&severity=INFORMATION"
+        resp = get_request(foglamp_url, get_url)
+        assert len(resp['audit'])
+        for audit_detail in resp['audit']:
+            if "test #1" == audit_detail['details']['name']:
```

Comment:
not the if based check, We need to ensure it exists. test will not fail if not found here.

Reply:
Here if is used because assert should run specifically for "test #1".I have added another assert to check the existence of "test #1" in updated code.